src/fondant/pipeline/pipeline.py

```diff
@@ -362,16 +379,38 @@ def read(
                 msg,
             )
 
-        operation = ComponentOp(
-            name_or_path,
-            produces=produces,
-            arguments=arguments,
-            input_partition_rows=input_partition_rows,
-            resources=resources,
-            cache=cache,
-            cluster_type=cluster_type,
-            client_kwargs=client_kwargs,
-        )
+        if issubclass(ref, BaseComponent):
+            if not ref.is_python_component():
+                err = """Only Python components are currently supported.
+                Make sure your class is decorated with the @python_component decorator."""
```

Comment:
This error message may be confusing, since we do accept non-Python components with this method, just not by direct reference.